librz/core/analysis_tp.c

```diff
@@ -535,44 +589,77 @@ RZ_API void rz_core_analysis_type_match(RzCore *core, RzAnalysisFunction *fcn) {
 			if (!i) {
 				rz_io_read_at(core->io, addr, buf, bsize);
 			}
-			ret = rz_analysis_op(analysis, &aop, addr, buf + i, bsize - i, RZ_ANALYSIS_OP_MASK_BASIC | RZ_ANALYSIS_OP_MASK_VAL);
-			if (ret <= 0) {
-				i += minopcode;
-				addr += minopcode;
-				rz_analysis_op_fini(&aop);
-				continue;
+			RzAnalysisOp *aop = NULL;
+			bool aop_cached = false;
+			size_t cache_len = rz_pvector_len(&op_cache);
+			if (cache_len > 0) {
+				aop = rz_pvector_at(&op_cache, cache_len - 1);
+				if (aop->addr == addr) {
+					aop_cached = true;
+				} else {
+					aop = NULL;
+				}
+			}
+			if (!aop) {
+				aop = rz_analysis_op_new();
+				if (!aop) {
+					break;
+				}
+			}
+			if (aop_cached) {
+				ret = aop->size;
+			} else {
+				ret = rz_analysis_op(analysis, aop, addr, buf + i, bsize - i, RZ_ANALYSIS_OP_MASK_BASIC | RZ_ANALYSIS_OP_MASK_VAL);
+				if (ret <= 0) {
+					i += minopcode;
+					addr += minopcode;
```

Comment:
shouldn't we add `addr` even when the data is taken from the cache?

Reply:
Ofc, it is done in the end of the loop body. BTW it is not my code.

Reply:
Oh i see, sorry